clang/lib/CIR/Lowering/ThroughMLIR/LowerCIRToMLIR.cpp

```diff
@@ -400,6 +400,44 @@ class CIRSinOpLowering : public mlir::OpConversionPattern<mlir::cir::SinOp> {
   }
 };
 
+class CIRBitClzOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::BitClzOp> {
+public:
+  using OpConversionPattern<mlir::cir::BitClzOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::BitClzOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto resultIntTy =
+        getTypeConverter()->convertType(op.getType()).cast<mlir::IntegerType>();
+    auto clz = rewriter.create<mlir::math::CountLeadingZerosOp>(
+        op->getLoc(), adaptor.getInput());
+    auto newOp = createIntCast(rewriter, clz->getResult(0), resultIntTy,
+                               /*isSigned=*/false);
+    rewriter.replaceOp(op, newOp);
+    return mlir::LogicalResult::success();
+  }
+};
+
+class CIRBitCtzOpLowering
+    : public mlir::OpConversionPattern<mlir::cir::BitCtzOp> {
+public:
+  using OpConversionPattern<mlir::cir::BitCtzOp>::OpConversionPattern;
+
+  mlir::LogicalResult
+  matchAndRewrite(mlir::cir::BitCtzOp op, OpAdaptor adaptor,
+                  mlir::ConversionPatternRewriter &rewriter) const override {
+    auto resultIntTy =
+        getTypeConverter()->convertType(op.getType()).cast<mlir::IntegerType>();
+    auto ctz = rewriter.create<mlir::math::CountTrailingZerosOp>(
+        op->getLoc(), adaptor.getInput());
+    auto newOp = createIntCast(rewriter, ctz->getResult(0), resultIntTy,
+                               /*isSigned=*/false);
+    rewriter.replaceOp(op, newOp);
+    return mlir::LogicalResult::success();
+  }
```

Comment:
These two `matchAndRewrite` functions looks similar. Could you refactor out a common `matchAndRewriteBitop` function to do the common job? You can reuse this function in later PRs when you try to lower other bit operations.

Reply:
Sounds good, I will do that right away.